backend/src/v5/processors/teamspaces/teamspaces.js

```diff
@@ -15,13 +15,31 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-const { getAccessibleTeamspaces } = require('../../models/users');
-const { getJobsToUsers } = require('../../models/jobs');
-const { getMembersInfo } = require('../../models/teamspaces');
+const { addDefaultJobs, assignUserToJob, getJobsToUsers } = require('../../models/jobs');
+const { createTeamspaceRole, grantTeamspaceRoleToUser } = require('../../models/roles');
+const { createTeamspaceSettings, getMembersInfo } = require('../../models/teamspaces');
+const { getAccessibleTeamspaces, grantTeamspacePermissionsToUser } = require('../../models/users');
+const { DEFAULT_JOBS } = require('../../models/jobs.constants');
 const { isTeamspaceAdmin } = require('../../utils/permissions/permissions');
+const { logger } = require('../../utils/logger');
 
 const Teamspaces = {};
 
+Teamspaces.initTeamspace = async (username) => {
+	try {
+		await createTeamspaceRole(username);
+		await Promise.all([
+			grantTeamspaceRoleToUser(username, username),
+			addDefaultJobs(username),
+			createTeamspaceSettings(username),
+			grantTeamspacePermissionsToUser(username, username, ['teamspace_admin']),
+		]);
+		await assignUserToJob(username, DEFAULT_JOBS.find((j) => j._id === 'Admin')._id, username);
```

Comment:
The find seems redundant

you're finding a default job that has this `_id`, and you're returning the _id.

probably better to have a constant as DEFAULT_OWNER_JOB = "Admin" in `jobs.constants` to avoid hardcoding admin here